### ISSUE TITLE
Unable to send email using office365 smtp servers

### ISSUE THREAD

**Issue by minhajuddin** [Vagabond/gen_smtp#114]
I want to use office365 smtp endpoint to send out emails. However, for some reason gen_smtp fails with an error:

```erlang
{:error, :retries_exceeded,
 {:temporary_failure, 'smtp.office365.com', :tls_failed}}
```

The code that I am trying to use is below:

```elixir
    :gen_smtp_client.send_blocking({'apisupport@mydomain.com',
                                   ['khaja@mailinator.com'],
                                   ('Subject: Awesome stuff \r\nFrom: apisupport@mydomain.com \r\nTo: khaja@mailinator.com \r\n\r\nThis is the email body'),
                                 },
      [
        {:relay, 'smtp.office365.com'},
        {:port, 587},
        {:hostname,'mydomain.com'},
        {:username, 'apisupport@mydomain.com'},
        {:password, 'password for username'},
        {:tls,:always},
        {:auth, :always},
      ])
```
I have all the required applications started when I check `Applications.started_applications`
```elixir
[{:email_debug, 'email_debug', '0.1.0'},
 {:gen_smtp, 'A generic Erlang SMTP server/client framework', '0.11.0'},
 {:ssl, 'Erlang/OTP SSL application', '8.1'},
 {:public_key, 'Public key infrastructure', '1.3'},
 {:asn1, 'The Erlang ASN1 compiler version 4.0.4', '4.0.4'},
 {:crypto, 'CRYPTO', '3.7.2'}, {:logger, 'logger', '1.3.4'},
 {:mix, 'mix', '1.3.4'}, {:elixir, 'elixir', '1.3.4'},
 {:compiler, 'ERTS  CXC 138 10', '7.0.3'}, {:stdlib, 'ERTS  CXC 138 10', '3.2'},
 {:kernel, 'ERTS  CXC 138 10', '5.1.1'}]
```

I have been trying to fix this for a couple of days without any headway.

I have tested it with ruby and it is working fine which rules out the possibility of auth or port issues.

```ruby
require 'pony'

Pony.mail(
  to: 'khaja@mailinator.com', from: 'apisupport@mydomain.com', subject: "Hello", body: "Hey bud",
  via: :smtp,
  via_options: {
    :address        => 'smtp.office365.com',
    :port           => '587',
    :user_name      => 'apisupport@mydomain.com',
    :enable_starttls_auto => true,
    :password       => 'password',
    :authentication => :login, # :plain, :login, :cram_md5, no auth by default
    :domain         => "mydomain.com" # the HELO domain provided by the client to the server
  }
)
```

**Comment by minhajuddin** [comment]
Looking at the wireshark traffic, I see an SSL request being sent even though it should be using the TLS protocol, as soon as gen_smtp sends an SSL request the server is sending a `FIN` which is causing the error. Not sure why we are sending an SSL request.

**Comment by minhajuddin** [comment]
I was able to do a quick fix by changing socket.erl by adding `{versions, ['tlsv1.2']},` Changing this makes the ssl code to use the TLS protocol instead of using SSL. I'll see If I can create a proper patch and send it.

```erlang
-define(SSL_CONNECT_OPTIONS,[ {active, false},
                              {depth, 0},
                              {packet, line},
                              {versions, ['tlsv1.2']},
                              {ssl_imp, new},
                              {ip, {0,0,0,0}},
                              {port, 0}]).

```

**Comment by minhajuddin** [comment]
Fixed by #115
